packages/mui-base/src/useMenu/menuReducer.ts

```diff
@@ -35,6 +35,13 @@ export default function menuReducer(
 
   if (action.type === ListActionTypes.blur) {
     if (!action.context.listboxRef.current?.contains(action.event.relatedTarget)) {
+      // check https://github.com/mui/material-ui/pull/36917#issuecomment-1566992698 for explaination
```

Comment:
```suggestion
      // To prevent the menu from closing when the focus leaves the menu to the button.
      // For more details, see https://github.com/mui/material-ui/pull/36917#issuecomment-1566992698
```